cmd/erasure-common.go

```diff
@@ -206,25 +206,30 @@ func (er erasureObjects) getLoadBalancedDisks(optimized bool) []StorageAPI {
 // This function does the following check, suppose
 // object is "a/b/c/d", stat makes sure that objects ""a/b/c""
 // "a/b" and "a" do not exist.
-func (er erasureObjects) parentDirIsObject(ctx context.Context, bucket, parent string) bool {
-	var isParentDirObject func(string) bool
-	isParentDirObject = func(p string) bool {
-		if p == "." || p == SlashSeparator {
+func (er erasureObjects) parentDirIsObject(ctx context.Context, bucket, parent string) (ret bool) {
+	segments := strings.Split(parent, slashSeparator)
+	if segments[0] == "" {
```

Comment:
you don't need this `if` check just do this 

```
for _, s := range segments {
     if s == "" {
           // No segments can be empty string, return false.
           break
     }
}

Reply:
Done